pkg/ebpf/netlink/conntracker.go

```diff
@@ -244,26 +246,31 @@ func (ctr *realConntracker) register(c ct.Con) int {
 		return 0
 	}
 
-	key, ok := formatKey(c)
-	if !ok {
-		return 0
-	}
+	registerTuple := func(keyTuple, transTuple *ct.IPTuple) {
+		key, ok := formatKey(keyTuple)
+		if !ok {
+			return
+		}
 
-	now := time.Now().UnixNano()
-	ctr.Lock()
-	defer ctr.Unlock()
+		now := time.Now().UnixNano()
 
-	if len(ctr.state) >= ctr.maxStateSize {
-		ctr.logExceededSize()
-		return 0
-	}
+		if len(ctr.state) >= ctr.maxStateSize {
+			ctr.logExceededSize()
+			return
+		}
 
-	generation := getNthGeneration(generationLength, now, 3)
-	ctr.state[key] = formatIPTranslation(c, generation)
+		generation := getNthGeneration(generationLength, now, 3)
+		ctr.state[key] = formatIPTranslation(transTuple, generation)
 
-	then := time.Now().UnixNano()
-	atomic.AddInt64(&ctr.stats.registers, 1)
-	atomic.AddInt64(&ctr.stats.registersTotalTime, then-now)
+		then := time.Now().UnixNano()
+		atomic.AddInt64(&ctr.stats.registers, 1)
+		atomic.AddInt64(&ctr.stats.registersTotalTime, then-now)
```

Comment:
Counting this way doesn't capture how long it takes to acquire the lock.  I think we can do the stats accounting outside of this function an count each connection as a single register. 

Same comment for unregister